pkgs/development/python-modules/pyarrow/default.nix

```diff
@@ -85,6 +98,27 @@ buildPythonPackage rec {
     export PYARROW_PARALLEL=$NIX_BUILD_CORES
   '';
 
+  postInstall = ''
+    # these are the steps that setting PYARROW_BUNDLE_ARROW_CPP_HEADERS=1 would
+    # take; doing it here allows us to avoid patching pyarrow's setup.py script to
+    # adjust the the permissions of pyarrow/include
```

Comment:
What permissions need to be adjusted? Is this a workaround for a bug?

Reply:
The `setup.py` file copies in directly from `arrow-cpp`, and that has different permissions from `pyarrow`

Reply:
Yeah, if you want to build against the pyarrow cpp API I think you need some way to access these headers, and if someone only runs `pip install pyarrow` then without bundled headers they won't be able to use the pyarrow cpp API.

Reply:
But we use `nix`, so this is not our problem :)

Reply:
Right :)

Reply:
So, let's drop this `postInstall` and add (optionally) `ARROW_ BUNDLE_ARROW_CPP_HEADERS = 0`?

Reply:
Ok, then we'll have to skip `test_cython.py` tests in pyarrow. These tests expect the headers to be bundled.

Reply:
Ah, okay. So we still need to keep the part of the `postInstall` that copies the pyarrow C++ API to the appropriate location, otherwise the pyarrow C++ API can't be used.

Reply:
I'm happy with this bit now. The upstream removal of the Python dependency from `arrow-cpp` simplified this whole bit.